mantidproject/mantid
# add ConfigService::remove to python API

### Description of work

#### Summary of work

Exposes `ConfigService::remove()` to the python API.

#### Purpose of work

SNAPRed needed to turn off mantid workbench logs while within workbench.  For reasons that remain dark and mysterious, this is only possible if mantid workbench has _no_ console channel set, such as when the properties file is deleted.

One way to achieve the same condition would be to remove the key.  Exposing this method allows for that to happen.

*There is no associated issue* but there are two associated ORNL EWMs
- [EWM 6257](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=6257) - the original story that presented this need
- [EWM 6854](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=6854) - the story to expose this property

#### Further detail of work

Just that single line.

### To test:

Build this branch.  Open workbench.  Run the below
``` python
from mantid.kernel import ConfigService
ConfigService.setString("logging.channels.consoleChannel.class", "PythonLoggingChannel")
```
Close workbench, then check `tail -10 ~/.mantid/Mantid.user.properties` and make sure the `consoleChannel` property is set.

Open workbench again.  Run the below
``` python
from mantid.kernel import ConfigService
ConfigService.remove("logging.channels.consoleChannel.class")
```
and close workbench.  Check the same file again.  Nothing should be there.

***BONUS POINTS** make sure this does what is intended withing SNAPRed.  You need to already have SNAPRed cloned from github.

First build this branch within your `mantid/build` folder, within the `mantid-developer` environment.

From within the build folder, run `./bin/AddPythonPath.py`

Navigate to the SNAPRed directory and pull [SNAPRed PR #443](https://github.com/neutrons/SNAPRed/pull/443).

Then 
``` bash
python -m pip install -e .
conda install pydantic
```
and finally
``` bash
python -m workbench
```

This will open the local workbench build with the SNAPRed branch.  Open the SNAPRed interace (under diffraction), and open the calibration panel.   Where it says "Run Number" enter any sequence of digits then click away.  On the workbench console, you will see some mantid errors.  Within the SNAPRed directory, open the file called mantidlog.txt.  It should show the same errors and also information about a call to `GetIPTS`.  This means this function was successful in allowing SNAPRed to split and redirect the mantid logs.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
